router/core/transport.go

```diff
@@ -84,68 +96,159 @@ func (ct *CustomTransport) RoundTrip(req *http.Request) (*http.Response, error)
 	return resp, err
 }
 
+func isRunningInsideDocker() bool {
```

Comment:
Please let us put this logic in a separate package so it doesn't bother us here too much. A simple test would be also great.